File: security/src/main/java/top/bettercode/summer/security/DefaultAuthority.java
package top.bettercode.summer.security;

import java.util.Arrays;
import java.util.Collection;
import java.util.Collections;
import java.util.HashSet;
import org.springframework.security.access.SecurityConfig;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.authority.SimpleGrantedAuthority;

/**
 * @author Peter Wu
 */
public class DefaultAuthority {

  public static GrantedAuthority DEFAULT_GRANTED_AUTHORITY = new SimpleGrantedAuthority(
      "authenticated");
  public static SecurityConfig DEFAULT_AUTHENTICATED = new SecurityConfig("authenticated");
  public static final SecurityConfig ROLE_ANONYMOUS = new SecurityConfig("ROLE_ANONYMOUS");

  public static boolean isDefaultAuthority(String authority) {
    return DEFAULT_GRANTED_AUTHORITY.getAuthority().equals(authority);
  }

  public static boolean isDefaultAuthority(GrantedAuthority authority) {
    return DEFAULT_GRANTED_AUTHORITY.equals(authority);
  }

  public static Collection<? extends GrantedAuthority> defaultAuthority() {
    return Collections.singleton(DEFAULT_GRANTED_AUTHORITY);
  }


  public static Collection<? extends GrantedAuthority> addDefaultAuthority(
      GrantedAuthority... authorities) {
    HashSet<GrantedAuthority> objects = new HashSet<>(Arrays.asList(authorities));
    objects.add(DEFAULT_GRANTED_AUTHORITY);
    return objects;
  }


  public static Collection<? extends GrantedAuthority> addDefaultAuthority(
      String... authorities) {
    HashSet<GrantedAuthority> objects = new HashSet<>();
    for (String authority : authorities) {
      objects.add(new SimpleGrantedAuthority(authority));
    }
    objects.add(DEFAULT_GRANTED_AUTHORITY);
    return objects;
  }

  public static Collection<? extends GrantedAuthority> addDefaultAuthority(
      Collection<String> authorities) {
    HashSet<GrantedAuthority> objects = new HashSet<>();
    for (String authority : authorities) {
      objects.add(new SimpleGrantedAuthority(authority));
    }
    objects.add(DEFAULT_GRANTED_AUTHORITY);
    return objects;
  }
}

File: security/src/main/kotlin/top/bettercode/summer/security/authorize/DefaultAuthority.kt
package top.bettercode.summer.security.authorize

import org.springframework.security.access.SecurityConfig
import org.springframework.security.core.GrantedAuthority
import org.springframework.security.core.authority.SimpleGrantedAuthority
import java.util.*

/**
 * @author Peter Wu
 */
object DefaultAuthority {
    const val DEFAULT_AUTHENTICATED_VALUE = "authenticated"

    @JvmField
    var DEFAULT_GRANTED_AUTHORITY: GrantedAuthority = SimpleGrantedAuthority(
            DEFAULT_AUTHENTICATED_VALUE)

    @JvmField
    var DEFAULT_AUTHENTICATED = SecurityConfig(DEFAULT_AUTHENTICATED_VALUE)

    @JvmField
    val ROLE_ANONYMOUS = SecurityConfig(Anonymous.ROLE_ANONYMOUS_VALUE)

    @JvmStatic
    fun isDefaultAuthority(authority: String): Boolean {
        return DEFAULT_GRANTED_AUTHORITY.authority == authority
    }

    @JvmStatic
    fun isDefaultAuthority(authority: GrantedAuthority): Boolean {
        return DEFAULT_GRANTED_AUTHORITY == authority
    }

    @JvmStatic
    fun defaultAuthority(): Collection<GrantedAuthority> {
        return setOf(DEFAULT_GRANTED_AUTHORITY)
    }

    @JvmStatic
    fun addDefaultAuthority(
            vararg authorities: GrantedAuthority
    ): Collection<GrantedAuthority> {
        val objects = HashSet(listOf(*authorities))
        objects.add(DEFAULT_GRANTED_AUTHORITY)
        return objects
    }

    @JvmStatic
    fun addDefaultAuthority(
            vararg authorities: String?
    ): Collection<GrantedAuthority> {
        val objects = HashSet<GrantedAuthority>()
        for (authority in authorities) {
            objects.add(SimpleGrantedAuthority(authority))
        }
        objects.add(DEFAULT_GRANTED_AUTHORITY)
        return objects
    }

    @JvmStatic
    fun addDefaultAuthority(
            authorities: Collection<String?>
    ): Collection<GrantedAuthority> {
        val objects = HashSet<GrantedAuthority>()
        for (authority in authorities) {
            objects.add(SimpleGrantedAuthority(authority))
        }
        objects.add(DEFAULT_GRANTED_AUTHORITY)
        return objects
    }
}
